pandas/tseries/offsets.py

```diff
@@ -2013,63 +1993,45 @@ def apply(self, other):
             datetime(other.year, self.startingMonth, 1))
         next_year = self.get_year_end(
             datetime(other.year + 1, self.startingMonth, 1))
+
         prev_year = tslib._localize_pydatetime(prev_year, other.tzinfo)
         cur_year = tslib._localize_pydatetime(cur_year, other.tzinfo)
         next_year = tslib._localize_pydatetime(next_year, other.tzinfo)
 
-        if n > 0:
-            if other == prev_year:
-                year = other.year - 1
-            elif other == cur_year:
-                year = other.year
-            elif other == next_year:
-                year = other.year + 1
-            elif other < prev_year:
-                year = other.year - 1
-                n -= 1
+        if other == prev_year:
+            n -= 1
+        elif other == cur_year:
+            pass
+        elif other == next_year:
+            n += 1
+            # TODO: Not hit in tests
```

Comment:
can we hit these in tests

Reply:
It wasn't obvious to me how to hit it, so I left the note to try again next time around.  (which will be soon.  We're far from done with offsets PRs)

Reply:
@jbrockmendel : If you don't mind explaining, how do you know that this particular branch isn't hit?  What about the other branches?

Reply:
BTW, the changes themselves look **OK** but would like to get some more color on the questions I just posed to you before merging.

Reply:
Put in a bunch of “assert false” statements as a quick coverage check.

Reply:
Ah, I see.  From which test files did those `assert false` failures then originate?

Reply:
Almost certainly `tests.tseries.offsets.test_offsets`

Reply:
Okay, sounds good.  Have a look at those tests once I merge to see how we can increase coverage.